### PR TITLE
Stop Collapsing NOT predicate on top of Quantified Subquery

### PR DESCRIPTION
**Pull request by xinzweb**
When there is no inverse operation available.

[#116788193]

### CODE REVIEW

File: data/dxl/minidump/CollapseNot.mdp [hunk 1]
```diff
@@ -0,0 +1,640 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<dxl:DXLMessage xmlns:dxl="http://greenplum.com/dxl/2010/12/">
```
Comment (vraghavan78) [inline data/dxl/minidump/CollapseNot.mdp]: schema and SQL please!

File: libgpopt/src/translate/CTranslatorDXLToExpr.cpp [hunk 2]
```diff
@@ -2619,6 +2620,13 @@ CTranslatorDXLToExpr::PexprCollapseNot
 		// get mdid and name of the inverse of the comparison operator used by quantified subquery
 		IMDId *pmdidOp = pdxlopSubqueryQuantified->PmdidScalarOp();
 		IMDId *pmdidInverseOp = m_pmda->Pmdscop(pmdidOp)->PmdidOpInverse();
+
+		if (NULL == pmdidInverseOp)
+		{
+			// collapsing NOT node failed
```
Comment (vraghavan78) [inline libgpopt/src/translate/CTranslatorDXLToExpr.cpp]: can we add a comment saying we do not have inverse operation, and therefore we are failing?

**Comment by vraghavan78** [comment]
+1

**Comment by hsyuan** [comment]
+1

File: data/dxl/minidump/CollapseNot.mdp (hunk 1 above)
Comment (xinzweb) [inline data/dxl/minidump/CollapseNot.mdp]: Fixed. Thanks.

**Pull request merged**